File: B/2182/2182.py
def contas(a, b):
    init = 1
    count = 0
    while a > 0 or b > 0:
        if a - init < 0:
            break
        a -= init
        count += 1
        temp = a
        a = b
        b = temp
        init *= 2
    return count

def main():
    n = int(input())
    for _ in range(n):
        a, b = map(int, input().split())
        op1 = contas(a, b)
        op2 = contas(b, a)
        print(max(op1, op2))

if __name__ == "__main__":
    main()

File: B/2182/2182.cpp
#include <iostream>

int contas(int a, int b) {
	int init = 1, count = 0, temp = 0;
	while (a > 0 || b > 0) {
		if (a - init < 0) break;
		a -= init;
		count++;
		temp = a;
		a = b;
		b = temp;
		init *= 2;
	}
	return count;
}
		
int main() {
	int n, a ,b;
	std::cin >> n;
	for (int i = 0; i < n; i++) {
		std::cin >> a >> b;
		int op1 = contas(a,b), op2 = contas(b,a);
		std::cout << std::max(op1,op2) << std::endl;
	}
	return 0;
}
